### DISCUSSION TITLE
Local plugin routes will not resolve to controller functions

### DISCUSSION THREAD

**Discussion by zwalkcodes** [strapi/strapi#6875] [Legacy Topics]
Hello,

I created a local plugin with the strapi generate::plugin method and I am having difficulty getting my routes to resolve to their corresponding controllers. I added these routes to the public user-permissions policy. I'm using latest 3.05 stable release on latest node as well. Any assistance is much appreciated. 

I hope strapi releases some more in depth documentation on plugin development soon. I've scoured through most everything I can find online to no avail. 

![Screen Shot 2020-06-30 at 9 28 03 PM](https://user-images.githubusercontent.com/19294209/86203045-c7910280-bb18-11ea-94c9-0777d60bd564.png)

![Screen Shot 2020-06-30 at 9 26 17 PM](https://user-images.githubusercontent.com/19294209/86203058-cf50a700-bb18-11ea-9562-b3592faa9e97.png)

**Comment by derrickmehaffy** [comment]
You need to return something for the controller to execute properly try adding `return { okay: true }`

**Reply by zwalkcodes** [reply]
I gave that a shot and still no go. It's weird because all of my API controllers outside of the plugins folder work as expected and I can strapi.log from them. Also if I add a strapi log outside of the export in this plugin controller I see it right before the server starts as expected too. I am seeing this error below in the browser console occasionally which I'm not totally sure about. Maybe related idk

![Screen Shot 2020-07-01 at 4 33 18 PM](https://user-images.githubusercontent.com/19294209/86300860-12faed80-bbb9-11ea-9b99-10fb51fdab1b.png)


![Screen Shot 2020-07-01 at 4 15 25 PM](https://user-images.githubusercontent.com/19294209/86300828-f5c61f00-bbb8-11ea-961e-1497c8164960.png)

![Screen Shot 2020-07-01 at 4 32 49 PM](https://user-images.githubusercontent.com/19294209/86300823-f2cb2e80-bbb8-11ea-9d16-68be728d1433.png)
